source/faq.txt

```diff
@@ -4,12 +4,176 @@ FAQ
 
 .. default-domain:: mongodb
 
-Frequently Asked Questions
-
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 1
+   :depth: 2
    :class: singlecol
 
-Lorem ipsum.
+General
+-------
+
+Why are there two versions of the classes? e.g. ``Mongo`` and ``MongoClient``?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``MongoDatabase``, ``MongoClient``, ``MongoCollection`` are the
+newer generation of classes and have better defaults for production use.
+
+For more information, see our blog post on `MongoClient
+<https://www.mongodb.com/blog/post/introducing-mongoclient>`_. 
+
+What's the difference between ``mongodb-driver-sync``, ``mongodb-driver`` or ``mongo-java-driver``?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :widths: 35 65
+   :header-rows: 1
+
+   * - Driver
+     - Description
+   * - ``mongodb-driver-sync``
+     - The synchronous driver **only**, new in 3.7. **No legacy classes.**
+   * - ``mongodb-driver-async``
+     - The new asynchronous driver **only**, new in 3.0. **No legacy classes.**
+   * - ``mongodb-driver``
+     - The synchronous driver plus the legacy driver, new in 3.0.
+   * - ``mongo-java-driver``
+     - An uber jar containing the bson library, the core library and the mongodb-driver. This artifact is a valid OSGi bundle.
+   * - ``mongodb-driver-core``
+     - The core library, (new in 3.0). **No BSON included (but is a dependency), and no high-level drivers.**
+
+POJOs
+-----
+
+Do I have to specify an ID field value myself?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine you have the following entity:
+
+.. code-block:: java
+
+   public class Entity {
+      public final ObjectId id;
+      public String value;
+
+      public Entity(){ }
+   }
+
+When inserting with ``collection.insertOne()``, the Codec will encode
+this into JSON, but skip the ``_id`` field since the value is ``null``.
```

Comment:
Note: the driver converts into `BSON` not JSON when talking to the server!

Since 3.10 the PojoCodec will automatically generate the ObjectId (https://jira.mongodb.org/browse/JAVA-2674) via the conventions.  

I think this section can be removed.